retail/interactive-tutorial/RetailSearch.Samples/SearchWithFacetSpec.cs

```diff
@@ -0,0 +1,116 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_search_product_with_facet_spec]
+
+using Google.Cloud.Retail.V2;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// Search with facet spec sample class.
+/// </summary>
+public static class SearchWithFacetSpec
+{
+    /// <summary>The project number.</summary>
+    private static readonly string ProjectNumber = Environment.GetEnvironmentVariable("PROJECT_NUMBER");
+
+    /// <summary>The default search placement.</summary>
+    private static readonly string DefaultSearchPlacement = $"projects/{ProjectNumber}/locations/global/catalogs/default_catalog/placements/default_search";
+
+    /// <summary>Get search request.</summary>
+    private static SearchRequest GetSearchRequest(string query, string facetKeyParam)
+    {
+        // Put the intervals here:
+        var facetKey = new SearchRequest.Types.FacetSpec.Types.FacetKey
+        {
+            Key = facetKeyParam
+        };
+
+        var facetSpec = new SearchRequest.Types.FacetSpec
+        {
+            FacetKey = facetKey
+        };
+
+        var searchRequest = new SearchRequest()
+        {
+            Placement = DefaultSearchPlacement, // Placement is used to identify the Serving Config name
+            Query = query,
+            VisitorId = "123456", // A unique identifier to track visitors
+            PageSize = 10
+        };
+
+        searchRequest.FacetSpecs.Add(facetSpec);
+
+        var jsonSerializeSettings = new JsonSerializerSettings
+        {
+            NullValueHandling = NullValueHandling.Ignore,
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+            Formatting = Formatting.Indented
+        };
+
+        var searchRequestJson = JsonConvert.SerializeObject(searchRequest, jsonSerializeSettings);
+
+        Console.WriteLine("Search. request: \n\n" + searchRequestJson);
+
+        return searchRequest;
+    }
+
+    /// <summary>Call the Retail Search.</summary>
+    [RetailSearch.Samples.Attributes.Example]
+    public static IEnumerable<SearchResponse> Search()
+    {
+        // Try different facets here:
+        string facetKey = "colorFamilies";
+        string query = "Tee";
+
+        var client = SearchServiceClient.Create();
+        var searchRequest = GetSearchRequest(query, facetKey);
+        var searchResponse = client.Search(searchRequest).AsRawResponses();
+
+        Console.WriteLine("\nSearch. response: \n");
+
+        var jsonSerializeSettings = new JsonSerializerSettings
```

Comment:
Again, it would be great if we could avoid all of this serialization. This is really unnecessary. Even for the tutorials there are better ways to show some of the results that match common use cases.

Reply:
Removed serialization and added hard-coded json elements in Request and Response by suggestion